### PR TITLE
Enable nullability in several UITypeEditors

### PR DESCRIPTION
**Pull request by halgab**
Those should be straightforward.

## Proposed changes

- Enable nullability in `ShortcutKeysEditor`
- Enable nullability in `DateTimeEditor`
- Enable nullability in `MaskedTextBoxTextEditor`

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/9611)

### CODE REVIEW

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/MaskedTextBoxTextEditor.cs [hunk 1]
```diff
@@ -2,24 +2,21 @@
...
 
 namespace System.Windows.Forms.Design;
 
 internal class MaskedTextBoxTextEditor : UITypeEditor
 {
-    public override object EditValue(ITypeDescriptorContext context, IServiceProvider provider, object value)
+    public override object? EditValue(ITypeDescriptorContext? context, IServiceProvider provider, object? value)
     {
-        if (context?.Instance is null | !provider.TryGetService(out IWindowsFormsEditorService editorService))
+        if (context?.Instance is null || !provider.TryGetService(out IWindowsFormsEditorService? editorService))
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/MaskedTextBoxTextEditor.cs:14]: nice catch!

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DateTimeEditor.DateTimeUI.cs [hunk 2]
```diff
@@ -17,7 +15,7 @@ public partial class DateTimeEditor
     private class DateTimeUI : Control
     {
         private readonly MonthCalendar _monthCalendar = new DateTimeMonthCalendar();
-        private IWindowsFormsEditorService _editorService;
+        private IWindowsFormsEditorService? _editorService;
```
[the remark this answers is not in the document]
Comment (halgab) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DateTimeEditor.DateTimeUI.cs]: I merged the functionality of `Start` in the constructor, and `End` in the disposing code.

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DateTimeEditor.DateTimeUI.cs [hunk 3]
```diff
@@ -19,22 +17,32 @@ private class DateTimeUI : Control
         private readonly MonthCalendar _monthCalendar = new DateTimeMonthCalendar();
         private IWindowsFormsEditorService _editorService;
 
-        public DateTimeUI()
+        public DateTimeUI(IWindowsFormsEditorService editorService, object? value)
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DateTimeEditor.DateTimeUI.cs:20]: this restructure, we can not take it to .NET8.0. please rebase to feature/9.0 branch and we can review there.

Comment (halgab) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DateTimeEditor.DateTimeUI.cs:20]: done

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DateTimeEditor.DateTimeUI.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (halgab) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DateTimeEditor.DateTimeUI.cs]: I already did that yesterday :)

**Comment by dreddy-work** [comment]
@halgab, can you get this rebased?

**Comment by halgab** [comment]
Sure. Should I rebase on [feature/9.0](https://github.com/dotnet/winforms/tree/feature/9.0) or main? And what about the other PRs currently pointing to [feature/9.0](https://github.com/dotnet/winforms/tree/feature/9.0)?

**Review by dreddy-work — APPROVED** [review]

**Pull request merged**